api/controllers/CheckUploadedDocumentsController.js

```diff
@@ -14,20 +17,32 @@ const CheckUploadedDocumentsController = {
             if (uploadedFileData.length === 0) {
                 throw new Error('No files uploaded');
             }
-            uploadedFileData.forEach(async (uploadedFile) => {
-                await UploadedDocumentUrls.create(
-                    CheckUploadedDocumentsController._dbColumnData(
-                        uploadedFile,
-                        req
+            for (let i = 0; i <= uploadedFileData.length; i++) {
+                const endOfLoop = i === uploadedFileData.length;
+                if (endOfLoop) {
+                    CheckUploadedDocumentsController._checkDocumentCountAndPaymentDetails(
+                        req,
+                        res
+                    );
+                } else {
```

Comment:
(This is minor, so no need to action) 

A guard clause can be used here like I mentioned above to unnest some of this logic. 
```suggestion
                if (endOfLoop) {
                    CheckUploadedDocumentsController._checkDocumentCountAndPaymentDetails(
                        req,
                        res
                    );
                    break //break keyword will "complete" the loop (exit it, like a return) 
                } 
```